src/main/groovy/com/cloudbees/plugins/flow/FlowDSL.groovy

```diff
@@ -457,19 +469,24 @@ public class FlowDelegate {
                     Result result = final_state.result
                     results.add(final_state)
                     current_state.result = current_state.result.combine(result)
-                } catch(ExecutionException e)
-                {
+                }
+                catch(ExecutionException e) {
                     // TODO perhaps rethrow?
                     current_state.result = FAILURE
                     listener.error("Failed to run DSL Script")
                     e.printStackTrace(listener.getLogger())
                 }
+                catch(CancellationException e) {
+                    current_state.result = ABORTED
+                }
             }
 
             pool.shutdown()
             pool.awaitTermination(1, TimeUnit.DAYS)
             current_state.lastCompleted =lastCompleted
         } finally {
+            pool.shutdown()
```

Comment:
please add a comment to explain why we are doing shutdown here and not waiting for task termination.